src/components/ha-button.ts

```diff
@@ -1,30 +1,171 @@
-import { Button } from "@material/mwc-button";
+import Button from "@shoelace-style/shoelace/dist/components/button/button.component";
+import styles from "@shoelace-style/shoelace/dist/components/button/button.styles";
 import { css } from "lit";
-import { customElement } from "lit/decorators";
-import { styles } from "@material/mwc-button/styles.css";
+import { customElement, property } from "lit/decorators";
 
+/**
+ * Home Assistant button component
+ *
+ * @element ha-button
+ * @extends {Button}
+ *
+ * @summary
+ * A stylable button component supporting Home Assistant theming, variants, and appearances.
+ *
+ * @slot - Label of the button
+ * @slot prefix - The prefix container (usually for icons).
+ * @slot suffix - The suffix container (usually for icons).
+ *
+ * @csspart base - The component's base wrapper.
+ * @csspart prefix - The container that wraps the prefix.
+ * @csspart label - The button's label.
+ * @csspart suffix - The container that wraps the suffix.
+ * @csspart caret - The button's caret icon, an `<sl-icon>` element.
+ * @csspart spinner - The spinner that shows when the button is in the loading state.
+ *
+ * @cssprop --ha-button-font-family - Font family for the button text.
+ * @cssprop --ha-font-weight-medium - Medium font weight for button text.
+ * @cssprop --ha-button-border-width - Border width for the button.
+ * @cssprop --ha-button-theme-color - Main color for the button.
+ * @cssprop --ha-button-theme-darker-color - Darker variant of the main color.
+ * @cssprop --ha-button-theme-lighter-color - Lighter variant of the main color.
+ * @cssprop --ha-button-height - Height of the button.
+ * @cssprop --ha-button-border-radius - Border radius for the button.
+ * @cssprop --ha-button-text-color - Text color for the button.
+ * @cssprop --ha-button-font-size - Font weight for the button text.
+ *
+ * @attr {("small"|"medium")} size - Sets the button size.
```

Comment:
Since we set every button to size small now, should we make small the default, so we dont have to set it everywhere?